### PR TITLE
[Keyboard] Added Roses Lite Support

### PR DESCRIPTION
**Pull request by vk2fgav**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Added Roses Lite Compatability

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/roseslite/config.h [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (fauxpark) [inline keyboards/roseslite/config.h:167]:
```suggestion
```

File: keyboards/roseslite/config.h [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+/* enable advanced MIDI features:
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
```
Comment (fauxpark) [inline keyboards/roseslite/config.h:204]:
```suggestion
```

File: keyboards/roseslite/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+* Keyboard Maintainer: [Reason6/Vk2fgav](https://github.com/vk2fgav)
+* Hardware Supported: Roses Series
+* Hardware Availability: [Shop](http://delta-design.xyz)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make roses_lite:default
+
+Flash example for this keyboard
+    make roses_lite:default:flash
```
Comment (fauxpark) [inline keyboards/roseslite/readme.md:13]:
```suggestion
Flashing example for this keyboard:

    make roseslite:default:flash
```

File: keyboards/roseslite/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Roses Lite
+
+
+* Keyboard Maintainer: [Reason6/Vk2fgav](https://github.com/vk2fgav)
+* Hardware Supported: Roses Series
+* Hardware Availability: [Shop](http://delta-design.xyz)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make roses_lite:default
```
Comment (fauxpark) [inline keyboards/roseslite/readme.md:10]:
```suggestion
    make roseslite:default
```

File: keyboards/roseslite/rules.mk [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/roseslite/rules.mk:24]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/roseslite/info.json [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+                {"label":"Shift", "x":1.5, "y":3, "w":2}, {"label":"~", "x":3.5, "y":3}, {"label":"Z", "x":4.5, "y":3}, {"label":"X", "x":5.5, "y":3}, {"label":"C", "x":6.5, "y":3}, {"label":"V", "x":7.5, "y":3},   {"label":"N", "x":10.25, "y":3}, {"label":"M", "x":11.25, "y":3}, {"label":"<", "x":12.25, "y":3}, {"label":">", "x":13.25, "y":3}, {"label":"?", "x":14.25, "y":3}, {"label":"Shift", "x":15.25, "y":3, "w":1.75},{"label":"UP", "x":17.5, "y":3}, {"label":"f6", "x":18.5, "y":2},
+                {"label":"f1", "x":0, "y":0}, {"label":"f1", "x":0, "y":1}, {"label":"f1", "x":0, "y":2}, {"label":"CTR", "x":1.5, "y":4, "w":1.25},{"label":"Win", "x":3.75, "y":4, "w":1.25},  {"label":"SPC", "x":5, "y":4, "w":2.75},{"label":"Alt", "x":7.75, "y":4}, {"label":"SPC","x":10.25, "y":4, "w":2.75}, {"label":"Alt", "x":13, "y":4, "w":1.25}, {"label":"L", "x":16.5, "y":4}, {"label":"D", "x":17.5, "y":4}, {"label":"R", "x":18.5, "y":4}
+            ]
+        }
+    }
+}
```
Comment (fauxpark) [inline keyboards/roseslite/info.json:18]:
```suggestion
{
    "keyboard_name": "roses lite",
    "url": "delta-design.xyz",
    "maintainer": "Reason6/Vk2fgav",
    "width": 18,
    "height": 5,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"x": 0, "y": 0},

                {"x": 1.5, "y": 0},
                {"x": 2.5, "y": 0},
                {"x": 3.5, "y": 0},
                {"x": 4.5, "y": 0},
                {"x": 5.5, "y": 0},
                {"x": 6.5, "y": 0},
                {"x": 7.5, "y": 0},

                {"x": 10, "y": 0},
                {"x": 11, "y": 0},
                {"x": 12, "y": 0},
                {"x": 13, "y": 0},
                {"x": 14, "y": 0},
                {"x": 15, "y": 0},
                {"x": 16, "y": 0},
                {"x": 17, "y": 0},

                {"x": 18.5, "y": 0},

                {"x": 0, "y": 1},

                {"x": 1.5, "y": 1, "w": 1.5},
                {"x": 3, "y": 1},
                {"x": 4, "y": 1},
                {"x": 5, "y": 1},
                {"x": 6, "y": 1},
                {"x": 7, "y": 1},

                {"x": 9.5, "y": 1},
                {"x": 10.5, "y": 1},
                {"x": 11.5, "y": 1},
                {"x": 12.5, "y": 1},
                {"x": 13.5, "y": 1},
                {"x": 14.5, "y": 1},
                {"x": 15.5, "y": 1},
                {"x": 16.5, "y": 1, "w": 1.5},

                {"x": 18.5, "y": 1},

                {"x": 0, "y": 2},

                {"x": 1.5, "y": 2, "w": 1.75},
                {"x": 3.25, "y": 2},
                {"x": 4.25, "y": 2},
                {"x": 5.25, "y": 2},
                {"x": 6.25, "y": 2},
                {"x": 7.25, "y": 2},

                {"x": 9.75, "y": 2},
                {"x": 10.75, "y": 2},
                {"x": 11.75, "y": 2},
                {"x": 12.75, "y": 2},
                {"x": 13.75, "y": 2},
                {"x": 14.75, "y": 2},
                {"x": 15.75, "y": 2, "w": 2.25},

                {"x": 18.5, "y": 2},

                {"x": 1.5, "y": 3, "w": 2},
                {"x": 3.5, "y": 3},
                {"x": 4.5, "y": 3},
                {"x": 5.5, "y": 3},
                {"x": 6.5, "y": 3},
                {"x": 7.5, "y": 3},

                {"x": 10.25, "y": 3},
                {"x": 11.25, "y": 3},
                {"x": 12.25, "y": 3},
                {"x": 13.25, "y": 3},
                {"x": 14.25, "y": 3},
                {"x": 15.25, "y": 3, "w": 1.75},

                {"x": 17.5, "y": 3},

                {"x": 1.5, "y": 4, "w": 1.25},
                {"x": 3.75, "y": 4, "w": 1.25},
                {"x": 5, "y": 4, "w": 2.75},
                {"x": 7.75, "y": 4},

                {"x": 10.25, "y": 4, "w": 2.75},
                {"x": 13, "y": 4, "w": 1.25},

                {"x": 16.5, "y": 4},
                {"x": 17.5, "y": 4},
                {"x": 18.5, "y": 4}
            ]
        }
    }
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by vk2fgav** [comment]
And again XD

**Review by fauxpark — APPROVED** [review]

**Comment by vk2fgav** [comment]
Honestly I love how a miss click doesnt have an undo button for something like reviews #thanksgit @fauxpark sorry to bother you again :/

**Review by vk2fgav — COMMENTED** [review]
@Gucci

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**